application-diagram-ui/src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -245,6 +245,26 @@ define('diagram-store', ['jquery', 'xwiki-meta', 'xwiki-utils', 'diagram-utils',
     left.done($.proxy(right, 'resolve')).fail($.proxy(right, 'reject'));
   };
 
+  var getScaleOfPNGImage = function() {
+    var scale = 1;
+    var doc = new XWiki.Document(xm.documentReference);
+    var spaces = doc.spaces.map(function(space) { return "spaces/" + space; });
+    var url = "/xwiki/rest/wikis/" + doc.wiki + "/" + spaces.join('/') + "/pages/" + doc.page
+        + "/objects/Diagram.DiagramClass/0/properties/pdfImageExportZoom";
+    $.ajax({
+      url: url,
+      async: false,
```

Comment:
Synchronous HTTP requests are not really recommended. Moreover I don't think you need to make an additional HTTP request. ``DiagramEditSheet`` [passes diagram configuration](https://github.com/xwikisas/application-diagram/blob/master/application-diagram-ui/src/main/resources/Diagram/DiagramEditSheet.xml#L81) through an attribute of the diagram element. The diagram configuration [is initialized](https://github.com/xwikisas/application-diagram/blob/master/application-diagram-ui/src/main/resources/Diagram/DiagramSheet.xml#L44) in ``DiagramSheet``. So you should put the value of ``pdfImageExportZoom`` there. Then, on the client side you just need to [add a require dependency](https://github.com/xwikisas/application-diagram/blob/master/application-diagram-ui/src/main/resources/Diagram/DiagramSheet.xml#L207) to the ``diagram-config`` module which will "inject" the configuration. ATM the configuration is "global" (two diagrams will share the same config on the client side) but that is OK for you because ``pdfImageExportZoom`` is used only in edit mode FTM, where you have a single diagram.

Reply:
So it means that pdfImageExportZoom will be the scale used in all exports, not specific to a diagram?

Reply:
AFAIU, the plan is to use ``pdfImageExportZoom`` **only** when generating the PNG image that is attached to the diagram page and later used to export the diagram to PDF. This PNG image is generated only from **edit mode**, when the diagram is saved. In edit mode there's a single diagram so the value of ``pdfImageExportZoom`` from the diagram configuration certainly comes from the edited diagram.

Reply:
Yes, it is just for the PNG image used in xwiki PDF export.
I just had the feeling that by using diagramConfig the pdfImageExportZoom should have been moved to DiagramConfigClass, meaning that it would be the same scale for all the images. But that is not the case.
All is clear now :)

Reply:
> I just had the feeling that by using diagramConfig the pdfImageExportZoom should have been moved to DiagramConfigClass

We could add the ``pdfImageExportZoom`` property to the ``DiagramConfigClass`` in the future, but I think it's still useful to be able to overwrite its value for a specific diagram, so having ``pdfImageExportZoom`` in ``DiagramClass`` for a start is not bad.